src/libraries/System.IO.Compression/src/System/IO/Compression/ZipArchive.cs

```diff
@@ -388,6 +388,11 @@ private ZipArchiveEntry DoCreateEntry(string entryName, CompressionLevel? compre
             if (_mode == ZipArchiveMode.Read)
                 throw new NotSupportedException(SR.CreateInReadMode);
 
+            if (_entriesDictionary.ContainsKey(entryName))
+            {
+                throw new InvalidOperationException(string.Format(SR.EntryNameAlreadyExists, entryName));
+            }
```

Comment:
I'm working on getting various MAUI workloads running on .NET 7. One of our tests hit an issue due to this change.

An MSBuild task inside a NuGet package has the following call:

https://github.com/xamarin/XamarinComponents/blob/895182cc16d88b0e092578d8a6d27bb68d0d459a/Util/Xamarin.Build.Download/source/Xamarin.Build.Download/AndroidAarFixups.cs#L60

Previously the code was creating a new entry then removing the old one afterward. (Code is old, it may indeed be incorrect)

Is the new exception intentional? Is it something that needs to be on a "breaking changes" list somewhere?

Reply:
I filed an issue; it seems I can reproduce in a smaller example: https://github.com/dotnet/runtime/issues/68734